server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -164,16 +160,12 @@ private boolean shouldSync(boolean didRefresh) {
         // is important to upload the zero state segments so that the restore does not break.
         return this.primaryTerm != indexShard.getOperationPrimaryTerm()
             || didRefresh
-            || remoteDirectory.getSegmentsUploadedToRemoteStore().isEmpty();
+            || remoteDirectory.getSegmentsUploadedToRemoteStore().isEmpty()
+            || isRefreshAfterCommitSafe();
```

Comment:
why do we need this check ?

Reply:
Added comments about this in the code itself.

Putting the same below - 
When the shouldSync is called the first time, then 1st condition on primary term is true. But after that we update the primary term and the same condition would not evaluate to true again in syncSegments. Below check ensures that if there is commit, then that gets picked up by both 1st and 2nd shouldSync call.